starfish/codebook.py

```diff
@@ -442,40 +456,45 @@ def metric_decode(
         Returns
         -------
         IntensityTable :
-            intensity table containing additional data variables for target assignments and metric
-            outputs
+            Intensity table containing normalized intensities, target assignments, distances to
+            the nearest code, and the filtering status of each feature.
 
         """
+
+        self._validate_decode_intensity_input_matches_codebook_shape(intensities)
+
         # normalize both the intensities and the codebook
         norm_intensities, norms = self._normalize_features(intensities, norm_order=norm_order)
         norm_codes, _ = self._normalize_features(self, norm_order=norm_order)
 
-        # mask low intensity features
-        intensity_mask = np.where(norms < min_intensity)[0]
-
         metric_outputs, targets = self._approximate_nearest_code(
```

Comment:
i don't remember if this was true in master or not. but we are taking a performance hit if we:

slower behavior
1. decode
2. threshold on intensity

versus.

faster behavior
1. threshold on intensity
2. decode

in the latter, we have less rows of data to pass to sklearn.nearestneighbors, which i believe is an n^2 operation.  on the other hand, the slower behavior does let us keep all distances computed for all pixels. 

my intuition  is that there should be an option in this function that lets you toggle between behavior 1 and behavior 2. perhaps this is outside the context for this PR -- but what do you think?

Reply:
Spoke offline, current implementation is OK due to use of approximate nearest neighbors algorithm, which is much faster than n^2. 

However, I created an issue to address this, as @dganguli correctly points out that we could make this much faster through intelligent pre-filtering  #532.